proxy_luafgen.c

```diff
@@ -26,16 +27,103 @@ int mcplib_funcgen_gc(lua_State *L) {
 // These "generator functions" handle pre-allocating and creating a memory
 // heirarchy, allowing dynamic runtimes at high speed.
 
+// must be called with fgen on top of stack in fgen->thread->L
+static void mcp_rcontext_cleanup(mcp_funcgen_t *fgen, mcp_rcontext_t *rctx, int fgen_idx) {
+    lua_State *L = fgen->thread->L;
+    luaL_unref(L, LUA_REGISTRYINDEX, rctx->coroutine_ref);
```

Comment:
I think it's better for general performance to get all of these references into the fgen local reference table.

Then there are numerous smaller tables instead of one gigantic lua registry, and those smaller tables get released on reload.

probably not worth doing in this commit, but I'll look in case it's easy and relatively safe.